substrate/frame/revive/src/tests.rs

```diff
@@ -4582,3 +4589,41 @@ fn tracing_works() {
 		}
 	});
 }
+
+#[test]
+fn unknown_precompiles_revert() {
+	let (code, _code_hash) = compile_module("read_only_call").unwrap();
+
+	ExtBuilder::default().build().execute_with(|| {
+		let _ = <Test as Config>::Currency::set_balance(&ALICE, 100_000_000_000);
+		let Contract { addr, .. } =
+			builder::bare_instantiate(Code::Upload(code)).build_and_unwrap_contract();
+
+		let cases: Vec<(H160, Box<dyn FnOnce(_)>)> = vec![
+			(
+				H160::from_low_u64_be(0x1),
```

Comment:
Are you absolutely sure that an address written in Solidity as `0x1` is meant as big endian number and not the byte string `0x01..00`?

Reply:
yeah
```rust
    assert_eq!(
        H160::from_low_u64_be(0x1),
        H160::from(hex_literal::hex!("0000000000000000000000000000000000000001"))
    );
```
I can add a test if that makes you feel better

Reply:
or just use the later form here maybe

Reply:
also see 
https://github.com/AcalaNetwork/Acala/blob/master/runtime/common/src/precompile/mod.rs?plain=1#L73

Reply:
> yeah
> 
> ```rust
>     assert_eq!(
>         H160::from_low_u64_be(0x1),
>         H160::from(hex_literal::hex!("0000000000000000000000000000000000000001"))
>     );
> ```
> 
> I can add a test if that makes you feel better

I want to know how this: `0x05` is interpreted:
https://github.com/OpenZeppelin/openzeppelin-contracts/blob/9586aaf35241daf4b17e4858bf7c86edbb4b7247/contracts/utils/math/Math.sol#L358

Will put the `0x05` into the first or the last byte. Your assert just demonstrated how big endian works. But not how we treat the arguments to `staticcall` are treated in Solidity (+revive). Probably need to have cyrill have a look at that.

Reply:
It's a number, the literal is "zero extended" in that sense.

Rust equivalent would be: `let foo: u64 = 1`

Reply:
`resolc` does then encode the number as big endian and pass a pointer to `pallet_revive`?

Reply:
> Are you absolutely sure that an address written in Solidity as `0x1` is meant as big endian number and not the byte string `0x01..00`?

Addresses are just 20 bytes, they don't have endianness. Address `0x1` expands into `0x0..1`.

Reply:
To recap:
- It's a number in YUL hence it will be little endian on PolkaVM
- But it should be Big Endian so we byte swap it before passing a pointer to it to pallet_revive
- Hence what we are doing here is correct

Reply:
Maybe it would help readability to write `H160::from(hex_literal::hex!("0000000000000000000000000000000000000001"))` instead. But any basic test should catch this anyways.

Reply:
Okay thanks. All clear then.